packages/dds/merge-tree/src/test/testClientLogger.ts

```diff
@@ -154,26 +159,63 @@ export class TestClientLogger {
         this.roundLogLines.push(this.localLine);
     }
 
-    public validate() {
-        const baseText = this.clients[0].getText();
+    public validate(opts?: {
+        clear?: boolean;
+        baseText?: string;
+        errorPrefix?: string;
+    }) {
+        const baseText = opts?.baseText ?? this.clients[0].getText();
+        const errorPrefix = opts?.errorPrefix ? `${opts?.errorPrefix}: ` : "";
+        // cache all the properties of client 0 for faster look up
+        const properties = Array.from({ length: this.clients[0].getLength() }).map(
+            (_, i) => this.clients[0].getPropertiesAtPosition(i));
         this.clients.forEach(
             (c) => {
-                if (c === this.clients[0]) { return; }
+                if (opts?.baseText === undefined && c === this.clients[0]) { return; }
                 // ensure all clients have seen the same ops
                 assert.equal(
                     c.getCurrentSeq(),
                     this.clients[0].getCurrentSeq(),
-                    `Client ${c.longClientId} current seq does not match client ${this.clients[0].longClientId}`,
+                    `${errorPrefix}${c.longClientId} current seq does not match client ${this.clients[0].longClientId}`,
                 );
                 // Pre-check to avoid this.toString() in the string template
                 if (c.getText() !== baseText) {
                     assert.equal(
                         c.getText(),
                         baseText,
                         // eslint-disable-next-line max-len
-                        `\n${this.toString()}\nClient ${c.longClientId} does not match client ${this.clients[0].longClientId}`);
+                        `${errorPrefix}\n${this.toString()}\nClient ${c.longClientId} does not match client ${opts?.baseText ? "baseText" : this.clients[0].longClientId}`);
                 }
+
+                if (c === this.clients[0]) { return; }
+                let pos = 0;
+                depthFirstNodeWalk(
```

Comment:
previously we didn't validate annotations. they are well tested independently, so this wasn't a huge deal, but this adds a bunch of coverage :)